CI + CD for browser builds

thomasballinger opened the pull request:
CI + (CD on pushes to master)

This requires a repo environment called "aws-deploy," which you can create empty to get the CI working.

thomasballinger committed:
CI + CD for browser builds

quyykk commented:
> This requires a repo environment called "aws-deploy," which you can create empty to get the CI working.

~~What do you mean by that?~~ nvm figured it out. thanks!

quyykk commented on .github/workflows/browser.yml:
```diff
@@ -0,0 +1,74 @@
+name: browser
+
+on:
+  push:
```
Do we need to add something here?

thomasballinger replied:
[This syntax seems right](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-multiple-events-with-activity-types-or-configuration) so syntax-wise I don't think so, but yes if the desired behavior is not to run on push to any branch.

> You must append a colon (:) to all events, including events without configuration.

quyykk replied:
> desired behavior is not to run on push

Now I'm confused. Why not?

thomasballinger replied:
Sorry, that was confusing. The clearer answer is "No, we do not need to add something here."

quyykk commented on .github/workflows/browser.yml:
```diff
@@ -0,0 +1,74 @@
+          aws-region: us-east-1
+
+      - name: Build
+        run: source emsdk/emsdk_env.sh; make output/index.html 
+      - name: Deploy
+        if: ${{ github.ref == 'refs/heads/master' && github.event_name == 'push' }}
+        run: source emsdk/emsdk_env.sh; make deploy
```
We can build in parallel by passing `-j $(nproc)` to make (to both calls).

quyykk reviewed (approved):
LGTM

thomasballinger committed:
use -j $(nproc)

thomasballinger committed:
fix Makefile

- using -j revealed missing dependencies
- use correct dataversion.js

thomasballinger committed:
don't invalidate play-endless-sky.com

It doesn't work for the editor anyway, I never set up the various cloudfront behaviors necessary.

thomasballinger committed:
up C++ version to cope with new changes on master
